### PR TITLE
Provide deprecation messages for `PackagePlugin.Path`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
As `PackagePlugin.Path` was deprecated in favor of `PackagePlugin.URL`, we should provide explicit deprecation messages that redirect plugin authors to the recommended use of the `URL` type.

rdar://128324920

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Review by bnbarham — APPROVED** [review]

File: Sources/PackagePlugin/Context.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ public struct PluginContext {
     /// write its outputs to that directory. The plugin may also create other
     /// directories for cache files and other file system content that either
     /// it or the command will need.
-    @available(_PackageDescription, deprecated: 6.0)
+    @available(_PackageDescription, deprecated: 6.0, renamed: "pluginWorkDirectory")
```
Comment (neonichu) [inline Sources/PackagePlugin/Context.swift:34]: renamed `pluginWorkDirectoryURL`

Comment (bnbarham) [reply Sources/PackagePlugin/Context.swift:34]: Ah nice catch, thanks @neonichu!

Comment (MaxDesiatov) [inline Sources/PackagePlugin/Context.swift:34]:
```suggestion
    @available(_PackageDescription, deprecated: 6.0, renamed: "pluginWorkDirectoryURL")
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**